packages/flutter_tools/test/general.shard/features_test.dart

```diff
@@ -137,6 +137,26 @@ void main() {
 
       expect(featureNames, unorderedEquals(testFeatureNames));
     });
+
+    testUsingContext('Feature runtime IDs are valid', () {
+      // Verify features' runtime IDs can be encoded into a Dart define.
+      final RegExp runtimeIdPattern = RegExp(r'^[a-zA-Z_]+$');
+      assert(runtimeIdPattern.hasMatch('multi_window'));
+      assert(!runtimeIdPattern.hasMatch('multi-window'));
+
+      for (final Feature feature in featureFlags.allFeatures) {
```

Comment:
Nit: consider filtering out features with null `runtimeId`s outside of the loop

```dart
for (final Feature feature in featureFlags.allFeatures.where((Feature f) => f.runtimeId != null)) {
  // ...
}
```

Reply:
I switch to the trick you suggested above instead: https://github.com/flutter/flutter/pull/168437/files/8121d41f6a157bf7ff6daeaaf3dbbffa0730ec15#r2124082944

Reply:
Nice! LGTM!